video/out/x11_common.c

```diff
@@ -626,6 +627,43 @@ static void vo_x11_get_x11_screen_dpi_scale(struct vo_x11_state *x11)
     }
 }
 
+// Get the dpi scale from the Xft.dpi resource. In practice, this value is much more
+// commonly used by GUI programs for scaling compared to the x11 screen dpi.
+// This is always a preference value so it's also more consistent.
+static bool vo_x11_get_xft_dpi_scale(struct vo_x11_state *x11)
+{
+    XrmInitialize();
+    char *resman = XResourceManagerString(x11->display);
+    if (!resman)
+        return false;
+
+    XrmDatabase db = XrmGetStringDatabase(resman);
+    if (!db)
+        return false;
+
+    XrmValue ret;
+    char *type;
+    double base_dpi = 96;
+    bool success = false;
+    XrmGetResource(db, "Xft.dpi", "String", &type, &ret);
+    if (ret.addr && !strcmp("String", type)) {
```

Comment:
what happens if `XrmGetResource` returns false and leaves the values uninitialized?

Reply:
> The `XrmValue` is set to NULL when `XrmGetResource` returns false

This is implementation details which can change (even if unlikely).

I don't see that it's documented to be NULL - https://tronche.com/gui/x/xlib/resource-manager/XrmQGetResource.html and https://linux.die.net/man/3/xrmqgetresource